pkg/cmd/roachtest/tests/failover.go

```diff
@@ -640,6 +715,26 @@ func makeFailer(
 	}
 }
 
+// Disconnect takes a set of nodes and each nodes internal ips. It disconnects
+// each node from all the others in the list.
+func Disconnect(t test.Test, c cluster.Cluster, ctx context.Context, nodes []int, reconnect bool) {
+	ips, err := c.InternalIP(ctx, t.L(), nodes)
+	require.NoError(t, err)
+
+	// disconnect each node from every other passed in node.
+	for n := 0; n < len(nodes); n++ {
+		for ip := 0; ip < len(ips); ip++ {
+			if n != ip {
+				if reconnect {
+					c.Run(ctx, c.Node(nodes[n]), `sudo iptables -F`)
+				} else {
+					c.Run(ctx, c.Node(nodes[n]), `sudo iptables -A INPUT -s `+ips[ip]+` -j DROP`)
```

Comment:
This is sort of an asymmetric partition too. It'll effectively prevent TCP connections from working since we block all return traffic, but for completeness we should add a corresponding `OUTPUT` rule too.